jujugui/static/gui/src/app/app.js

```diff
@@ -1656,6 +1626,50 @@ YUI.add('juju-gui', function(Y) {
       }
     },
 
+    createSocketURL: function(apiServer, apiPort, uuid) {
+      var currentHost = window.location.hostname,
+          currentPort = window.location.port;
+
+      var protocol = this.get('socket_protocol') || 'wss';
+      var baseUrl = protocol + '://' + currentHost;
+      if (currentPort !== '') {
+        baseUrl += ':' + currentPort;
+      }
+      baseUrl += '/ws';
+
+      if (!apiServer || !apiPort) {
+        var apiAddress = window.juju_config.apiAddress.replace('wss://', '');
+        apiAddress = apiAddress.split(':');
+        apiServer = apiAddress[0];
+        apiPort = apiAddress[1];
+      }
+
+      //XXX j.c.sackett 2015-12-18 We want to default to the bare /ws url in
```

Comment:
I think we can drop support for Juju < 1.20, clearly document it, and never use the old /ws path again.